app/src/main/java/ar/scacchipa/twittercloneapp/fragment/FragmentLoginApi.kt

```diff
@@ -4,21 +4,42 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
 import android.webkit.WebViewClient
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import ar.scacchipa.twittercloneapp.R
 import ar.scacchipa.twittercloneapp.databinding.FragmentLoginApiLayoutBinding
+import ar.scacchipa.twittercloneapp.viewmodel.LoginApiViewModel
 
 
 class FragmentLoginApi : Fragment() {
 
+    private val viewModel: LoginApiViewModel by viewModels()
     private var binding: FragmentLoginApiLayoutBinding? = null
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
+
+        val clientId = getString(R.string.client_id)
+        val redirectUri = getString(R.string.redirect_uri)
+
         binding = FragmentLoginApiLayoutBinding.inflate(inflater)
 
+        viewModel.userAccessToken.observe(viewLifecycleOwner) {
+            if (it.access_token != "") {
+                val action = FragmentLoginApiDirections
+                    .actionFragmentLoginApiToFragmentHappyPath(it.access_token)
+                findNavController().navigate(action)
+            } else {
+                findNavController().navigate(R.id.action_fragmentLoginApi_to_fragmentSadPath)
+            }
+        }
+
         binding?.loginApiWebview?.let { webView ->
             webView.settings.javaScriptEnabled = true
```

Comment:
here maybe you can use somekind of kotlin function that help you reduce code duplication for this field , like with or apply

Reply:
'let' was replaced with 'apply'